test/index.js

```diff
@@ -460,4 +470,81 @@ describe('KeyringController', function () {
       );
     });
   });
+
+  describe('submitEncryptionKey', function () {
+    it('sets encryption key data upon submitPassword', async function () {
+      keyringController.cacheEncryptionKey = true;
+      await keyringController.submitPassword(password);
+
+      expect(keyringController.password).toBe(password);
+      expect(keyringController.encryptionSalt).toBe('SALT');
+    });
+
+    it('unlocks the keyrings with valid information', async function () {
+      keyringController.cacheEncryptionKey = true;
+      const returnValue = await keyringController.encryptor.decryptWithKey();
+      const stub = sinon.stub(keyringController.encryptor, 'decryptWithKey');
+      stub.resolves(Promise.resolve(returnValue));
+
+      keyringController.store.updateState({ vault: MOCK_ENCRYPTION_DATA });
+      await keyringController.submitEncryptionKey(MOCK_ENCRYPTION_KEY);
+
+      expect(keyringController.encryptor.decryptWithKey.calledOnce).toBe(true);
+      expect(keyringController.keyrings).toHaveLength(1);
+    });
+
+    it('should not load keyrings when incorrect encryptionKey', async function () {
+      keyringController.cacheEncryptionKey = true;
+      keyringController.setLocked(true);
+
+      await expect(keyringController.submitEncryptionKey(`{}`)).rejects.toThrow(
```

Comment:
We could use an additional test where a _valid_ but _incorrect_ encyption key is passed in. I wouldn't call `{}` an "incorrect" key, it's not a valid key, making this more of a validation test.

Reply:
Done!